src/qibo/tests/test_core_circuit_noise.py

```diff
@@ -160,3 +160,31 @@ def test_density_matrix_circuit_errors():
     c = Circuit(5)
     with pytest.raises(ValueError):
         c.add(gates.KrausChannel([((0,), np.eye(2))]))
+
+
+def test_circuit_add_sampling():
+    """Check measurements when simulating added circuits with noise"""
+    # Create random noisy circuit and add noiseless inverted circuit
+    gates_set = [gates.X, gates.Y, gates.Z, gates.H, gates.S, gates.SDG, gates.I]
+    circ = Circuit(1)
+    circ_no_noise = Circuit(1)
+    for _ in range(100):
+        new_gate = np.random.choice(gates_set)(0)
+        circ.add(gates.PauliNoiseChannel(0, pz=0.01))
+        circ.add(new_gate)
+        circ_no_noise.add(new_gate)
+        circ.add(gates.PauliNoiseChannel(0, pz=0.01))
```

Comment:
I've implemented a test that checks the sampling after fixing the seed (using `K.set_seed(123)`.
However, I've notice that the parameter `seed` of the `PauliNoiseChannel` does not affect the results of the sampling. That's why here I've decided to omit it. Is this the correct behaviour, @stavros11 ?

Reply:
Thanks for adding the test.

> However, I've notice that the parameter `seed` of the `PauliNoiseChannel` does not affect the results of the sampling. That's why here I've decided to omit it. Is this the correct behaviour, @stavros11 ?

Can you give an example where it doesn't affect? For example, the following script:
```Python
from qibo import models, gates
import numpy as np


circ = models.Circuit(1)
circ_no_noise = models.Circuit(1)

for _ in range(10):
    new_gate = gates.H(0)
    circ.add(gates.PauliNoiseChannel(0, pz=0.1))
    circ.add(new_gate)
    circ_no_noise.add(new_gate)

circ.add(gates.PauliNoiseChannel(0, pz=0.1))
circ += circ_no_noise.invert()
circ.add(gates.M(0))

print(circ(nshots=100).samples()[:, 0])
```
gives me different samples every time I run, while if I add a `seed` in the last noise channel, it gives me the same samples every time.